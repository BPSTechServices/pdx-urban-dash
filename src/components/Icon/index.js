import React from 'react';
import PropTypes from 'prop-types';
import {
  Target,
  TrendingUp,
  TrendingDown,
  Maximize,
  Minimize,
  GitHub,
  ChevronsDown,
  ChevronsUp,
  XCircle,
  HelpCircle,
} from 'react-feather';

import { colors, iconSizes } from '../../constants';

const Icon = ({ size, type }) => {
  if (typeof type === 'object') return null;
  const validSizes = {
    xs: iconSizes.xsmall,
    sm: iconSizes.small,
    md: iconSizes.medium,
    lg: iconSizes.large,
    xl: iconSizes.xlarge,
  };

  const mSize = Object.keys(validSizes).includes(size) ? size : 'md';

  switch (type.toLowerCase()) {
    case 'target':
      return <Target size={validSizes[mSize]} />;
    case 'off-target':
      return <Target size={validSizes[mSize]} color={colors.warning} />;
    case 'on-target':
      return <Target size={validSizes[mSize]} color={colors.success} />;
    case 'trending-up':
      return <TrendingUp size={validSizes[mSize]} />;
    case 'trending-down':
      return <TrendingDown size={validSizes[mSize]} />;
    case 'maximize':
      return <Maximize size={validSizes[mSize]} />;
    case 'minimize':
      return <Minimize size={validSizes[mSize]} />;
    case 'github':
      return <GitHub size={validSizes[mSize]} />;
    case 'chevrons-down':
      return <ChevronsDown size={validSizes[mSize]} />;
    case 'chevrons-up':
      return <ChevronsUp size={validSizes[mSize]} />;
    case 'x-circle':
      return <XCircle size={validSizes[mSize]} />;
    case 'help-circle':
      return <HelpCircle size={validSizes[mSize]} />;
    default:
      throw new Error(`Icon type: ${type} does not exist`);
  }
};

Icon.propTypes = {
  size: PropTypes.string,
  type: PropTypes.string.isRequired,
};

Icon.defaultProps = {
  size: 'md',
};

export default Icon;
