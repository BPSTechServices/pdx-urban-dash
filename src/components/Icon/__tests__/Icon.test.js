import React from 'react';
import { mount } from 'enzyme';

import Icon from '../index';

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="target" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="off-target" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="on-target" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="TrendingUp" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="TrendingDown" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="Maximize" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="Minimize" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Icon />', () => {
  test('Icon renders target', () => {
    const wrapper = mount(<Icon type="Github" />);
    expect(wrapper).toMatchSnapshot();
  });
});
    
    
  
