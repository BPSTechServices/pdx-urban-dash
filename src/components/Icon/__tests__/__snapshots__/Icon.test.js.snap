// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Icon renders target 1`] = `ReactWrapper {}`;

exports[`<Icon /> Icon renders target 2`] = `ReactWrapper {}`;

exports[`<Icon /> Icon renders target 3`] = `ReactWrapper {}`;

exports[`<Icon /> Icon renders target 4`] = `ReactWrapper {}`;

exports[`<Icon /> Icon renders target 5`] = `ReactWrapper {}`;

exports[`<Icon /> Icon renders target 6`] = `ReactWrapper {}`;

exports[`<Icon /> Icon renders target 7`] = `ReactWrapper {}`;

exports[`<Icon /> Icon renders target 8`] = `ReactWrapper {}`;
